### ISSUE TITLE
vogels_sprekeler_synapse / sign check

### ISSUE THREAD

**Issue by paulbrodersen** [nest/nest-simulator#825]
The Vogels-Sprekeler synapse implements a constant weight decay for each pre-synaptic spike. 
For sufficient pre-synaptic spikes, this can result in the weight becoming 0, which is expected and desired behaviour . However, if the initial weight was negative, nest throws an error if subsequently the synapse properties are changed (e.g. when the learning rate is changed). 

MWE: (kinda)

```python 
import nest

a, b = [[gid] for gid in nest.Create('iaf_psc_alpha', 2)]
nest.Connect(a, b,
             conn_spec={'rule':'one_to_one'},
             syn_spec={'model'  :'vogels_sprekeler_synapse',
                       'weight' :-0.,
                       'Wmax'   :-1.})
```

I would appreciate any pointers on how to work around this issue.

**Comment by paulbrodersen** [comment]
My cpp is rusty and was never any good to begin with. However, I think that the problem is in vogels_sprekeler_connection.h, line 302 and below:

 ```cpp
      if ( not( ( ( weight_ >= 0 ) - ( weight_ < 0 ) )
             == ( ( Wmax_   >= 0 ) - ( Wmax_   < 0 ) ) ) )
      {
          throw BadProperty( "Weight and Wmax must have same sign." );
      }
```

Presumably, escaping the check if `weight_` is 0 should do the trick, i.e:

```cpp
if ( not( ( ( weight_ >= 0 ) - ( weight_ < 0 ) )
         == ( ( Wmax_ >= 0 ) - ( Wmax_ < 0 ) ) ) )
  {
    throw BadProperty( "Weight and Wmax must have same sign." );
  }

```

**Comment by sanjayankur31** [comment]
It's been a while since I wrote this, but from the lines you've quoted, it seems that we classify the weight either as negative (inhibitive in NEST convention) or non negative (rather than positive). You're getting the error because of the initial weight that you're setting - `-0.` does not qualify as negative. I set it to a very small negative quantity when I use it - `-0.00000000001` or the sort, to ensure that it matches the sign of `Wmax`. 

We could improve the check as you suggest, but i'll have to give it a little more thought to be sure that it not have any other consequences. 

@nest/nest-simulator-developers  - could someone assign this to me please?

**Comment by paulbrodersen** [comment]
I chose a bad example. I actually encounter the error when I have a negative weight that decays to zero. As there is no Wmin parameter, there is currently no way of preventing such a decay to zero, at which point the error is triggered.
